lib/genome/online_updaters/docm/updater.rb

```diff
@@ -0,0 +1,132 @@
+require 'net/http'
+module Genome; module OnlineUpdaters; module Docm
+  class Updater
+    attr_reader :new_version
+    def initialize(source_db_version = Date.today.strftime("%d-%B-%Y"))
+      @new_version = source_db_version
+    end
+
+    def update
+      remove_existing_source
+      source = create_new_source
+      create_interaction_claims(source)
+    end
+
+    private
+    def create_interaction_claims(source)
+      api_client = ApiClient.new
+      api_client.variants.each do |variant|
+        interaction_information = parse_interaction_information(variant)
+        interaction_information.each do |interaction_info|
+          gc = get_or_create_gene_claim(variant, source)
+          dc = get_or_create_drug_claim(interaction_info, source)
+          get_or_create_interaction_claim(gc, dc, interaction_info, variant['diseases'], source)
+        end
+      end
+    end
+
+    def get_or_create_gene_claim(variant, source)
+      DataModel::GeneClaim.where(
+        source_id: source.id,
+        nomenclature: 'DoCM Entrez Gene Symbol',
+        name: variant['gene']
+      ).first_or_create
+      #Entrez Gene Id
+    end
+
+    def parse_interaction_information(variant)
+      if !variant.key?('meta')
+        return []
+      end
+      fields = variant['meta'].first['Drug Interaction Data']['fields']
+      rows = variant['meta'].first['Drug Interaction Data']['rows']
+      interaction_information = Set.new()
+      rows.each do |row|
+        row_hash = {}
+        fields.zip(row).each do |name, value|
+          row_hash[name] = value
+        end
+        row_hash['Therapeutic Context'].split(/,|\+|plus/).each do |drug|
+          info = row_hash
+          info['Therapeutic Context'] = drug
```

Comment:
My concern here was that if I end up overriding the actual value of `row_hash['Therapeutic Context']` while looping over the same value, that it would somehow screw up the looping. ...although it gets run through `split` so maybe I don't need to be concerned after all? I still think I would prefer to avoid overwriting the original `row_hash` entry so using `dup` would probably be the cleanest solution.